src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs

```diff
@@ -1307,6 +1307,113 @@ static void Main()
         }
         #endregion
 
+        #region CallerArgumentExpression - Test various symbols
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestIndexers()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class Program
+{
+    const string i = nameof(i);
+    public int this[int i, [CallerArgumentExpression(i)] string s = ""<default-arg>""]
+    {
+        get => i;
+        set => Console.WriteLine($""{i}, {s}"");
+    }
+
+    public static void Main()
+    {
+        new Program()[1+  1] = 5;
+        new Program()[2+  2, ""explicit-value""] = 5;
+    }
+}
+";
+
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            CompileAndVerify(compilation, expectedOutput: @"2, 1+  1
+4, explicit-value").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestDelegate()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class Program
+{
+    const string s1 = nameof(s1);
+    delegate void D(string s1, ref string s2, out string s3, [CallerArgumentExpression(s1)] string s4 = """");
+
+    static void M(string s1, ref string s2, out string s3, string s4 = """")
+    {
+        s3 = """";
+        Console.WriteLine($""s1: {s1}"");
+        Console.WriteLine($""s2: {s2}"");
+        Console.WriteLine($""s3: {s3}"");
+        Console.WriteLine($""s4: {s4}"");
+    }
+
+    public static void Main()
+    {
+        D d = M;
+        string s2 = ""s2-arg"";
+        d.Invoke(""s1-arg"", ref s2, out _);
+        //d.EndInvoke(ref s2, out _, null);
```

Comment:
>EndInvoke [](http://example.com/codeflow?start=12&length=9)

We need to test interesting scenarios and the API behavior for EndInvoke too.

Reply:
@AlekseyTs My issue with testing any behavior with EndInvoke is that it throws PlatformNotSupportedException. I'm not aware of the history here, but I tried to target .NET Framework, and have the attribute manually written in source, but it still throws the exception. Any pointer on what to do here regarding testing?

Reply:
> My issue with testing any behavior with EndInvoke is that it throws PlatformNotSupportedException. I'm not aware of the history here, but I tried to target .NET Framework, and have the attribute manually written in source, but it still throws the exception. Any pointer on what to do here regarding testing?

If we cannot execute the code we can stop at observing IL and behavior of the API.